qiskit/providers/aer/backends/cluster/clusterjobset.py

```diff
@@ -0,0 +1,236 @@
+            AerClusterTimeoutError: if unable to retrieve all job results before the
+                specified timeout.
+        """
+        if self._results is not None:
+            return self._results
+
+        start_time = time.time()
+        original_timeout = timeout
+
+        # We'd like to use futures.as_completed or futures.wait
+        #   however this excludes the use of dask as executor
+        #   because dask's futures are not ~exactly~ the same.
+        res = []
+        for cjob in self._futures:
+            try:
+                result = cjob.result(timeout=timeout, raises=raises)
+                if result is None or not result.success:
+                    if result:
+                        logger.warning('ClusterJob %s Error: %s', cjob.name(), result.header)
+                    else:
+                        logger.warning('ClusterJob %s did not return a result', cjob.name())
+                res.append(result)
+            except AerClusterTimeoutError:
+                raise AerClusterTimeoutError(
+                    'Timeout while waiting for the results of experiment {}'.format(
+                        cjob.name()))
+            if timeout:
+                timeout = original_timeout - (time.time() - start_time)
+                if timeout <= 0:
+                    raise AerClusterTimeoutError(
+                        "Timeout while waiting for JobSet results")
+
+        self._results = CResults(self, res)
+
+        return self._results
+
+    @requires_submit
+    def cancel(self) -> None:
+        """Cancel all jobs in this job set."""
+        for cjob in self._futures:
+            cjob.cancel()
+
+    @requires_submit
+    def job(self, experiment: Union[str, QuantumCircuit, Schedule]) -> Tuple[CJob, int]:
+        """Retrieve the job used to submit the specified experiment and its index.
+
+        Args:
+            experiment: Retrieve the job used to submit this experiment. Several
+                types are accepted for convenience:
+
+                    * str: The name of the experiment.
+                    * QuantumCircuit: The name of the circuit instance will be used.
+                    * Schedule: The name of the schedule instance will be used.
+
+        Returns:
+            A tuple of the job used to submit the experiment and the experiment index.
+
+        Raises:
+            AerClusterJobNotFound: If the job for the experiment could not
+                be found.
+        """
+        if isinstance(experiment, (QuantumCircuit, Schedule)):
+            experiment = experiment.name
+        for job in self.jobs():
+            for i, exp in enumerate(job.qobj().experiments):
+                if hasattr(exp.header, 'name') and exp.header.name == experiment:
+                    return job, i
+
+        raise AerClusterJobNotFound(
+            'Unable to find the job for experiment {}.'.format(experiment))
+
+    @requires_submit
+    def jobs(self) -> List[Union[CJob, None]]:
```

Comment:
See above comment about `worker_job` function. If a worked job failed to submit I think it should raise an exception rather than return None.